packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -74,6 +74,10 @@ const BlockToolbar = ( { hideDragHandle } ) => {
 		};
 	}, [] );
 
+	const isReusable =
+		blockType.name === 'core/block' && blockType.category === 'reusable';
+	const isTemplatePart = blockType.name === 'core/template-part';
```

Comment:
There are some functions `isTemplatePart` and `isReusableBlock` in the blocks package that could be useful for these checks.